### ISSUE TITLE
Homebridge can't be install

### ISSUE THREAD

**Issue by Clemdu63** [nfarina/homebridge#1683]
Hello I'm here to tell you that I can't install homebridge in my synology ds215J with v6.11.2 of node.js

I have this error in terminal : 

`npm install -g --unsafe-perm homebridge
/volume1/@appstore/Node.js_v6/usr/local/bin/homebridge -> /volume1/@appstore/Node.js_v6/usr/local/lib/node_modules/homebridge/bin/homebridge

> mdns@2.3.4 install /volume1/@appstore/Node.js_v6/usr/local/lib/node_modules/homebridge/node_modules/mdns
> node-gyp rebuild

gyp ERR! build error 
gyp ERR! stack Error: not found: make
gyp ERR! stack     at getNotFoundError (/usr/local/lib/node_modules/npm/node_modules/which/which.js:14:12)
gyp ERR! stack     at F (/usr/local/lib/node_modules/npm/node_modules/which/which.js:69:19)
gyp ERR! stack     at E (/usr/local/lib/node_modules/npm/node_modules/which/which.js:81:29)
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/which/which.js:90:16
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/which/node_modules/isexe/index.js:44:5
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/which/node_modules/isexe/access.js:8:5
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:123:15)
gyp ERR! System Linux 3.2.40
gyp ERR! command "/volume1/@appstore/Node.js_v6/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /volume1/@appstore/Node.js_v6/usr/local/lib/node_modules/homebridge/node_modules/mdns
gyp ERR! node -v v6.11.2
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
/volume1/@appstore/Node.js_v6/usr/local/lib
└── (empty)

npm ERR! Linux 3.2.40
npm ERR! argv "/volume1/@appstore/Node.js_v6/usr/local/bin/node" "/usr/local/bin/npm" "install" "-g" "--unsafe-perm" "homebridge"
npm ERR! node v6.11.2
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE

npm ERR! mdns@2.3.4 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the mdns@2.3.4 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the mdns package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs mdns
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls mdns
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /root/npm-debug.log
npm ERR! code 1
`

**Comment by nlappas** [comment]
same here with debian

**Comment by aswathraja** [comment]
@nlappas did you install the dependencies : Avahi and dnss 

> sudo apt-get install libavahi-compat-libdnssd-dev

**Comment by Clemdu63** [comment]
Yes I have install all things

**Comment by nlappas** [comment]
I am so sorry. I tried to install with the npm that comes with debian stables, which is apparently way too old. I downloaded and installed manually the newest nodejs, and now homebridge works properly :) 
I apologize for pinging you

**Comment by Clemdu63** [comment]
how are you doing that ?

**Comment by nlappas** [comment]
@Clemdu63 Here is an instructions page: [https://nodejs.org/en/download/package-manager/](https://nodejs.org/en/download/package-manager/)

**Comment by Clemdu63** [comment]
I don't understand when I do the first command of instructions 
`curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -`
 I have :
```
root@Famille:~# curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -

## Installing the NodeSource Node.js v8.x repo...


## Populating apt-get cache...

+ apt-get update
bash: apt-get: command not found
Error executing command, exiting
root@Famille:~# 

```

**Comment by nlappas** [comment]
What linux distribution do you have?
run this:
```bash
cat /etc/*-release
```

**Comment by Clemdu63** [comment]
```
root@Famille:/# cat /etc/*-release
PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
NAME="Debian GNU/Linux"
VERSION_ID="8"
VERSION="8 (jessie)"
ID=debian
HOME_URL="http://www.debian.org/"
SUPPORT_URL="http://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
root@Famille:/# 

```

**Comment by nlappas** [comment]
Weird. `apt` should come installed with debian.

Try the command `dpkg`. If it returns command not found, then you will have to fix you distro before trying to install anything (let alone homebridge).

If the `dpkg` command works and returns an error, then you can download the `apt` manually and install it. http://forums.debian.net/viewtopic.php?f=5&t=1056

http://forums.debian.net/viewtopic.php?f=5&t=1056

**Comment by Clemdu63** [comment]
```
root@Famille:/# dpkg
dpkg: error: need an action option

Type dpkg --help for help about installing and deinstalling packages [*];
Use 'apt' or 'aptitude' for user-friendly package management;
Type dpkg -Dhelp for a list of dpkg debug flag values;
Type dpkg --force-help for a list of forcing options;
Type dpkg-deb --help for help about manipulating *.deb files;

Options marked [*] produce a lot of output - pipe it through 'less' or 'more' !
root@Famille:/# 

```

**Comment by nlappas** [comment]
ok then move forward and install apt as explained here http://forums.debian.net/viewtopic.php?f=5&t=1056.

Make sure that the version number in the installation command matches the one you download

**Comment by Clemdu63** [comment]
when I do that I have this : 

```
root@Famille:/# dpkg -i apt_1.4.8_amd64.deb
dpkg: error processing archive apt_1.4.8_amd64.deb (--install):
 package architecture (amd64) does not match system (armhf)
Errors were encountered while processing:
 apt_1.4.8_amd64.deb
root@Famille:/# 
```
so how I can know what is good for my system ?

**Comment by nlappas** [comment]
Oh now I noticed that you do not try to install it on a PC.
I have no clue then whether all of these packages are compiled for your CPU architecture. You will have to research online how to install homebridge for synology.
Have you seen someone succeeding and you tried?

**Comment by Clemdu63** [comment]
docker is not compatible with my synology nas I follow this tutorial but when I try to install homebridge I have the error of my subject ^^

This is the tutorial I follow 

https://medium.com/@gregwym/setup-homebridge-on-synology-nas-without-docker-f42edd174d0e

**Comment by nlappas** [comment]
The error says that you do not even have `make` installed

```bash
gyp ERR! build error 
gyp ERR! stack Error: not found: make

```
Did you follow the instructions?
Did you do the `apt-get install build-essential` ?

**Comment by Clemdu63** [comment]
Yes I have do that but look when I do the following instructions I have that : 

```
root@Famille:/# wget https://nodejs.org/dist/latest-v8.x/
converted 'https://nodejs.org/dist/latest-v8.x/' (ANSI_X3.4-1968) -> 'https://nodejs.org/dist/latest-v8.x/' (UTF-8)
--2018-01-02 19:50:01--  https://nodejs.org/dist/latest-v8.x/
Resolving nodejs.org (nodejs.org)... 104.20.22.46, 104.20.23.46, 2400:cb00:2048:1::6814:162e, ...
Connecting to nodejs.org (nodejs.org)|104.20.22.46|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: 'index.html'

index.html              [ <=>                  ]   4.98K  --.-KB/s   in 0.001s 

2018-01-02 19:50:03 (3.53 MB/s) - 'index.html' saved [5095]

root@Famille:/# tar -zxvf latest-v8.x/
tar (child): latest-v8.x/: Cannot open: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
root@Famille:/# ./configure
bash: ./configure: No such file or directory
root@Famille:/# 

```

**Comment by nlappas** [comment]
your command just downloaded the webpage (`index.html`) not nodejs.

**Comment by Clemdu63** [comment]
Yes I'm stupid ;)

I have to download with one ?
../
docs/                                              08-Dec-2017 15:27                   -
win-x64/                                           07-Dec-2017 22:22                   -
win-x86/                                           07-Dec-2017 22:06                   -
SHASUMS256.txt                                     08-Dec-2017 23:44                3671
SHASUMS256.txt.asc                                 08-Dec-2017 23:44                4208
SHASUMS256.txt.sig                                 08-Dec-2017 23:44                 310
node-v8.9.3-aix-ppc64.tar.gz                       08-Dec-2017 14:51            21842332
node-v8.9.3-darwin-x64.tar.gz                      08-Dec-2017 15:20            16039251
node-v8.9.3-darwin-x64.tar.xz                      08-Dec-2017 15:21            10431576
node-v8.9.3-headers.tar.gz                         08-Dec-2017 15:29              498036
node-v8.9.3-headers.tar.xz                         08-Dec-2017 15:29              361200
node-v8.9.3-linux-arm64.tar.gz                     08-Dec-2017 22:47            17757195
node-v8.9.3-linux-arm64.tar.xz                     08-Dec-2017 22:49            10732624
node-v8.9.3-linux-armv6l.tar.gz                    08-Dec-2017 23:16            17021497
node-v8.9.3-linux-armv6l.tar.xz                    08-Dec-2017 23:31            10269000
node-v8.9.3-linux-armv7l.tar.gz                    08-Dec-2017 14:19            17035236
node-v8.9.3-linux-armv7l.tar.xz                    08-Dec-2017 14:22            10257992
node-v8.9.3-linux-ppc64le.tar.gz                   08-Dec-2017 14:14            17747581
node-v8.9.3-linux-ppc64le.tar.xz                   08-Dec-2017 14:14            10608360
node-v8.9.3-linux-s390x.tar.gz                     08-Dec-2017 14:07            18300662
node-v8.9.3-linux-s390x.tar.xz                     08-Dec-2017 14:08            10958756
node-v8.9.3-linux-x64.tar.gz                       08-Dec-2017 14:10            17901223
node-v8.9.3-linux-x64.tar.xz                       08-Dec-2017 14:11            11395380
node-v8.9.3-linux-x86.tar.gz                       08-Dec-2017 14:09            17431387
node-v8.9.3-linux-x86.tar.xz                       08-Dec-2017 14:11            11001396
node-v8.9.3-sunos-x64.tar.gz                       08-Dec-2017 14:27            19376494
node-v8.9.3-sunos-x64.tar.xz                       08-Dec-2017 14:28            11953000
node-v8.9.3-sunos-x86.tar.gz                       08-Dec-2017 14:23            18072377
node-v8.9.3-sunos-x86.tar.xz                       08-Dec-2017 14:23            11135608
node-v8.9.3-win-x64.7z                             08-Dec-2017 15:19             9018231
node-v8.9.3-win-x64.zip                            08-Dec-2017 15:20            16894327
node-v8.9.3-win-x86.7z                             08-Dec-2017 15:14             7923609
node-v8.9.3-win-x86.zip                            08-Dec-2017 15:14            15391350
node-v8.9.3-x64.msi                                08-Dec-2017 15:20            16674816
node-v8.9.3-x86.msi                                08-Dec-2017 15:14            15085568
node-v8.9.3.pkg                                    08-Dec-2017 15:04            16308477
node-v8.9.3.tar.gz                                 08-Dec-2017 15:23            31121503
node-v8.9.3.tar.xz

**Comment by nlappas** [comment]
Try the `node-v8.9.3-linux-armv6l.tar.gz 08-Dec-2017 23:16 17021497`
or `node-v8.9.3-linux-armv7l.tar.gz 08-Dec-2017 14:19 17035236`

**Comment by Clemdu63** [comment]
ok thanks and after how I have to enter the "unpacked node dir" ?

**Comment by nlappas** [comment]
`cd`  and the name of the directory you want to got to

**Comment by Clemdu63** [comment]
I can't put `cd` dir ?

**Comment by nlappas** [comment]
You need to do some reading on unix. http://www.ee.surrey.ac.uk/Teaching/Unix/unix1.html

Sorry

**Comment by Clemdu63** [comment]
yes ok I have my directory
`root@Famille:/volume1# `
 but when I do  
`./configure`
I have still
```
root@Famille:/volume1# ./configure
bash: ./configure: No such file or directory
root@Famille:/volume1# 
```

**Comment by Clemdu63** [comment]
@nlappas any idea ?

**Comment by nlappas** [comment]
You are not inside the extracted directory. You need to get in there first and then execute the './configure'.

Instead of wasting your time with trial and error spend some time to learn the fundamentals of Unix.

**Comment by Clemdu63** [comment]
I have read fundamentals of Unix and now look I'm in the extracted directory : but nothings happens

```
root@Famille:/volume1# ls
@nodejs.core  node-v8.9.3-linux-armv6l	       node-v8.9.3-linux-armv6l.tar.gz.1
@optware      node-v8.9.3-linux-armv6l.tar.gz
root@Famille:/volume1# 
root@Famille:/volume1# cd /volume1/node-v8.9.3-linux-armv6l/
root@Famille:/volume1/node-v8.9.3-linux-armv6l# ./configure
bash: ./configure: No such file or directory
root@Famille:/volume1/node-v8.9.3-linux-armv6l# ls
CHANGELOG.md  LICENSE  README.md  bin  include	lib  share
root@Famille:/volume1/node-v8.9.3-linux-armv6l# 

```

**Comment by nlappas** [comment]
Then if it says that there is no configure file there is no configure file. It means that you try to apply blindly the instructions in a folder that does not match the instructions.

The node distribution that you downloaded (`node-v8.9.3-linux-armv6l.tar.gz`) has node precompiled for you. You just go in to the /bin directory and execute node from there. (You will see 3 executables in there `node`, `npm`, `npx`)

If you want to compile node from source code, as you see in the instructions, then you download the source https://nodejs.org/dist/latest-v9.x/node-v9.3.0.tar.gz This will have your configure file and you can try to compile it. Apart from the fact that this will take long time, there is high probability that you will get stuck due to missing dependencies. 

But the fact that you cannot have node running in a non-standard architecture machine is not an issue that you should be posting in the homebridge issue tracker.

**Comment by Clemdu63** [comment]
can you help me just for that please ? When I do the ./configure I have this "WARNING: C++ compiler too old, need g++ 4.9.4 or clang++ 3.4.2 (CXX=g++)"

**Comment by nlappas** [comment]
google is your friend

https://forum.synology.com/enu/viewtopic.php?t=95716

**Comment by Clemdu63** [comment]
nope because I have not IPKG and I can't install it

**Comment by nlappas** [comment]
You have to have a working package manager. Either `apt-get` or `ipkg` or `yum`. If you do not have you have to install one somehow. There are guides online, you can search.

The other option requires you  to build everything from source. Even GCC. And GCC compilation takes one hour on a i7 quad core with 16GB of memory. For your system probably its impossible to build GCC due to memory restrictions. You will have to crosscompile from another machine and transfer the compiled binaries and libraries.

In other words: make sure that you have a package manager working and install everything precompiled from there.

The alternative of compilation from source on a Synology machine is a rough and lonely route.

Good luck.

**Comment by Clemdu63** [comment]
nope because for my synology nobody know how to install ipkg for it so I have to install homebridge without ipkg so it's why I have ask the question to "how install homebridge with this synology ?!"

**Comment by ebaauw** [comment]
I don’t have a Synology, but I understood that you cannot install homebridge on it natively.  It runs inside a Docker container.
